### PR TITLE
Basic Twitter adapter

### PR DESCRIPTION
**Pull request by kevinnio**
This adds a simple twitter adapter including its spec file. This adapter allows the app to fetch tweets from Twitter and translates them into an internal representation of a `Post`.

### CODE REVIEW

File: crawler/adapters/twitter.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      client.search("#{hashtag}", result_type: :recent).take(amount).map do |tweet|
+        Post.new tweet.text, build_url(tweet.url), :twitter
+      end
+    end
+
+    def build_url(url)
+      "#{url.scheme}://#{url.host}#{url.path}"
+    end
+
+    def client
```
Comment (vzqzac) [inline crawler/adapters/twitter.rb]: I think this being `private` would be better

File: crawler/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# frozen_string_literal: true
+
+Post = Struct.new :content, :url, :origin
```
Comment (vzqzac) [inline crawler/models/post.rb:3]: ❤️

**Review by vzqzac — CHANGES_REQUESTED** [review]
Just a small change and what Edwin said about hiding the ENV vars, aside of that, good work! (still not a fan of the commits look though 😛)

File: crawler/adapters/twitter.rb (hunk 1 above)
Comment (kevinnio) [reply crawler/adapters/twitter.rb]: You're right. Will change.

File: crawler/adapters/twitter.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+module Adapters
+  class Twitter < Base
+    def fetch(hashtag:, amount:)
+      client.search("#{hashtag}", result_type: :recent).take(amount).map do |tweet|
+        Post.new tweet.text, build_url(tweet.url), :twitter
+      end
+    end
+
+    def build_url(url)
```
Comment (vzqzac) [inline crawler/adapters/twitter.rb]: I think we can move this to the `private` block too unless you're using it outside of here

**Review by vzqzac — CHANGES_REQUESTED** [review]

**Review by vzqzac — APPROVED** [review]
Good work!

**Pull request merged**